docs/Developer-guide.md

```diff
@@ -0,0 +1,131 @@
+
+The `main()` function is likely to be the same in each script. The alternative could be to create a script class with pre-implemented `main()` method. But each script could become a little less readable for beginner users wanting to run and edit those scripts.
+
+Each script should be runnable on its own, and provide a clear set of usage instructions when using `-h` option.
+
+## Scripts inputs, outputs and parameters
+
+Each script `get_arg_parser()` function builds an `argparse` for the script which details the inputs, outputs and parameters.
+
+**For inputs and outputs, we propose to use directories exclusively**. The goal is to be compatible with pipeline orchestration platforms (like AzureML) where inputs can be either file paths or directory paths, depending on the platform.
+
+For this, each script input will be considered a distinct directory containing at least one file (several if needed). Each script output will be provided as a directory in which the script will create one or multiple files on its own.
+
+Two helper functions in the common library will help to handle that type of i/o arguments consistently through the benchmark.
+
+## Common libraries
+
+We'll build common libraries under `/src/common/` to help script developers to achieve common routines in a standard and consistent manner.
+
+For instance:
+- mlflow metrics recording
+- managing typical arguments (ex: input/output folders with single file)
+
+## Python code template
+
+Here's a proposed python script template.
+
+```python
+# <<<any imports>>>
+
+# let's add the right PYTHONPATH for common module
+# (assuming your script is located at /src/scripts/SOMETHING/foo.py)
+COMMON_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..'))
+
+if COMMON_ROOT not in sys.path:
+    print(f"Adding {COMMON_ROOT} to PYTHONPATH")
+    sys.path.append(str(COMMON_ROOT))
+
+# before doing local import
+from common.metrics import MetricsLogger
+from common.io import input_file_path
+
+def get_arg_parser(parser=None):
+    """Adds component/module arguments to a given argument parser.
+
+    Args:
+        parser (argparse.ArgumentParser): an argument parser instance
+
+    Returns:
+        ArgumentParser: the argument parser instance
+
+    Notes:
+        if parser is None, creates a new parser instance
+    """
+    # add arguments that are specific to the module
+    if parser is None:
+        parser = argparse.ArgumentParser(__doc__)
+
+    # <<< whatever parsing building is needed>>>
+
+    return parser
+
+
+def run(args, unknown_args=[]):
+    """Run script with arguments (the core of the component)
+
+    Args:
+        args (argparse.namespace): command line arguments provided to script, see get_arg_parser()
+        unknown_args (list[str]): list of arguments not known by parser
+    """
+    # <<<whatever needs to be implemented here>>>
+    pass
+
+
+def main(cli_args=None):
```

Comment:
Are scripts expected to set up logging / reporting here?

Reply:
Yes, added as an issue #13